buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidLibTarget.groovy

```diff
@@ -14,8 +17,18 @@ class AndroidLibTarget extends AndroidTarget {
 
     private static final String KOTLIN_EXTENSIONS_OPTION = "plugin:org.jetbrains.kotlin.android:"
 
+    final AndroidLibInstrumentationTarget libInstrumentationTarget
+
     AndroidLibTarget(Project project, String name, boolean isTest = false) {
         super(project, name, isTest)
+
+        TestExtension testExtension = rootProject.okbuck.test
+        if (testExtension.espresso && instrumentationTestVariant && ProjectUtil.getType(project) == ProjectType.ANDROID_LIB) { // do not try to create this for android apps
```

Comment:
This adds a lot of performance overhead for projects with a lot of library projects and no instrumentation tests on libraries.

I would recommend gating this entire functionality behind an okbuck extension flag like

```
okbuck {
  test {
    espressoForLibraries = true
  }
}
```